glibc >= 2.42 makes memchr() const-correct, so assigning its result to char * fails under upstream's -Werror
--- cowsql-1.15.9.orig/src/server.c	2025-07-11 12:18:16.000000000 +0200
+++ cowsql-1.15.9/src/server.c	2026-09-12 23:56:08.094235933 +0200
@@ -1017,7 +1017,7 @@ static int parseNodeStore(char *buf, siz
 	struct client_node_info info;
 
 	version_str = p;
-	nl = memchr(p, '\n', (size_t)(end - version_str));
+	nl = (char *)memchr(p, '\n', (size_t)(end - version_str));
 	if (nl == NULL) {
 		return 1;
 	}
@@ -1029,7 +1029,7 @@ static int parseNodeStore(char *buf, siz
 
 	while (p != end) {
 		addr = p;
-		nl = memchr(p, '\n', (size_t)(end - addr));
+		nl = (char *)memchr(p, '\n', (size_t)(end - addr));
 		if (nl == NULL) {
 			return 1;
 		}
@@ -1037,7 +1037,7 @@ static int parseNodeStore(char *buf, siz
 		p = nl + 1;
 
 		id_str = p;
-		nl = memchr(p, '\n', (size_t)(end - id_str));
+		nl = (char *)memchr(p, '\n', (size_t)(end - id_str));
 		if (nl == NULL) {
 			return 1;
 		}
@@ -1056,7 +1056,7 @@ static int parseNodeStore(char *buf, siz
 		}
 
 		role_str = p;
-		nl = memchr(p, '\n', (size_t)(end - role_str));
+		nl = (char *)memchr(p, '\n', (size_t)(end - role_str));
 		if (nl == NULL) {
 			return 1;
 		}
@@ -1153,7 +1153,7 @@ static int parseLocalInfo(char *buf,
 	unsigned long long id;
 
 	version_str = p;
-	nl = memchr(version_str, '\n', (size_t)(end - version_str));
+	nl = (char *)memchr(version_str, '\n', (size_t)(end - version_str));
 	if (nl == NULL) {
 		return 1;
 	}
@@ -1164,7 +1164,7 @@ static int parseLocalInfo(char *buf,
 	}
 
 	addr = p;
-	nl = memchr(addr, '\n', (size_t)(end - addr));
+	nl = (char *)memchr(addr, '\n', (size_t)(end - addr));
 	if (nl == NULL) {
 		return 1;
 	}
@@ -1172,7 +1172,7 @@ static int parseLocalInfo(char *buf,
 	p = nl + 1;
 
 	id_str = p;
-	nl = memchr(id_str, '\n', (size_t)(end - id_str));
+	nl = (char *)memchr(id_str, '\n', (size_t)(end - id_str));
 	if (nl == NULL) {
 		return 1;
 	}
